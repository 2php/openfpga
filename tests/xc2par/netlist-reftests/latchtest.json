{
  "nodes": {
    "storage": [
      {
        "variant": {
          "AndTerm": {
            "inputs_true": [
              {
                "i": 2,
                "type_marker": null
              }
            ],
            "inputs_comp": [],
            "output": {
              "i": 3,
              "type_marker": null
            }
          }
        },
        "name": "$auto$coolrunner2_fixup.cc:101:execute$305",
        "location": null
      },
      {
        "variant": {
          "Xor": {
            "orterm_input": null,
            "andterm_input": {
              "i": 3,
              "type_marker": null
            },
            "invert_out": false,
            "output": {
              "i": 4,
              "type_marker": null
            }
          }
        },
        "name": "$auto$coolrunner2_fixup.cc:108:execute$306",
        "location": null
      },
      {
        "variant": {
          "AndTerm": {
            "inputs_true": [
              {
                "i": 5,
                "type_marker": null
              },
              {
                "i": 6,
                "type_marker": null
              }
            ],
            "inputs_comp": [],
            "output": {
              "i": 7,
              "type_marker": null
            }
          }
        },
        "name": "$auto$coolrunner2_sop.cc:143:execute$274",
        "location": null
      },
      {
        "variant": {
          "AndTerm": {
            "inputs_true": [],
            "inputs_comp": [
              {
                "i": 8,
                "type_marker": null
              }
            ],
            "output": {
              "i": 9,
              "type_marker": null
            }
          }
        },
        "name": "$auto$coolrunner2_sop.cc:143:execute$277",
        "location": null
      },
      {
        "variant": {
          "AndTerm": {
            "inputs_true": [],
            "inputs_comp": [
              {
                "i": 6,
                "type_marker": null
              },
              {
                "i": 10,
                "type_marker": null
              }
            ],
            "output": {
              "i": 11,
              "type_marker": null
            }
          }
        },
        "name": "$auto$coolrunner2_sop.cc:143:execute$280",
        "location": null
      },
      {
        "variant": {
          "AndTerm": {
            "inputs_true": [],
            "inputs_comp": [
              {
                "i": 5,
                "type_marker": null
              },
              {
                "i": 10,
                "type_marker": null
              }
            ],
            "output": {
              "i": 12,
              "type_marker": null
            }
          }
        },
        "name": "$auto$coolrunner2_sop.cc:143:execute$282",
        "location": null
      },
      {
        "variant": {
          "AndTerm": {
            "inputs_true": [
              {
                "i": 6,
                "type_marker": null
              },
              {
                "i": 2,
                "type_marker": null
              }
            ],
            "inputs_comp": [],
            "output": {
              "i": 13,
              "type_marker": null
            }
          }
        },
        "name": "$auto$coolrunner2_sop.cc:143:execute$287",
        "location": null
      },
      {
        "variant": {
          "AndTerm": {
            "inputs_true": [],
            "inputs_comp": [
              {
                "i": 5,
                "type_marker": null
              },
              {
                "i": 6,
                "type_marker": null
              }
            ],
            "output": {
              "i": 14,
              "type_marker": null
            }
          }
        },
        "name": "$auto$coolrunner2_sop.cc:143:execute$290",
        "location": null
      },
      {
        "variant": {
          "AndTerm": {
            "inputs_true": [],
            "inputs_comp": [
              {
                "i": 2,
                "type_marker": null
              }
            ],
            "output": {
              "i": 15,
              "type_marker": null
            }
          }
        },
        "name": "$auto$coolrunner2_sop.cc:143:execute$293",
        "location": null
      },
      {
        "variant": {
          "Xor": {
            "orterm_input": null,
            "andterm_input": {
              "i": 7,
              "type_marker": null
            },
            "invert_out": false,
            "output": {
              "i": 8,
              "type_marker": null
            }
          }
        },
        "name": "$auto$coolrunner2_sop.cc:154:execute$275",
        "location": null
      },
      {
        "variant": {
          "Xor": {
            "orterm_input": null,
            "andterm_input": {
              "i": 9,
              "type_marker": null
            },
            "invert_out": false,
            "output": {
              "i": 16,
              "type_marker": null
            }
          }
        },
        "name": "$auto$coolrunner2_sop.cc:154:execute$278",
        "location": null
      },
      {
        "variant": {
          "Xor": {
            "orterm_input": null,
            "andterm_input": {
              "i": 13,
              "type_marker": null
            },
            "invert_out": false,
            "output": {
              "i": 17,
              "type_marker": null
            }
          }
        },
        "name": "$auto$coolrunner2_sop.cc:154:execute$288",
        "location": null
      },
      {
        "variant": {
          "Xor": {
            "orterm_input": null,
            "andterm_input": {
              "i": 15,
              "type_marker": null
            },
            "invert_out": false,
            "output": {
              "i": 18,
              "type_marker": null
            }
          }
        },
        "name": "$auto$coolrunner2_sop.cc:154:execute$294",
        "location": null
      },
      {
        "variant": {
          "OrTerm": {
            "inputs": [
              {
                "i": 12,
                "type_marker": null
              },
              {
                "i": 11,
                "type_marker": null
              }
            ],
            "output": {
              "i": 19,
              "type_marker": null
            }
          }
        },
        "name": "$auto$coolrunner2_sop.cc:217:execute$284",
        "location": null
      },
      {
        "variant": {
          "Xor": {
            "orterm_input": {
              "i": 19,
              "type_marker": null
            },
            "andterm_input": null,
            "invert_out": false,
            "output": {
              "i": 20,
              "type_marker": null
            }
          }
        },
        "name": "$auto$coolrunner2_sop.cc:223:execute$285",
        "location": null
      },
      {
        "variant": {
          "InBuf": {
            "output": {
              "i": 10,
              "type_marker": null
            },
            "schmitt_trigger": false,
            "termination_enabled": false,
            "uses_data_gate": false
          }
        },
        "name": "$auto$iopadmap.cc:369:execute$295",
        "location": {
          "fb": 0,
          "i": 13
        }
      },
      {
        "variant": {
          "InBuf": {
            "output": {
              "i": 21,
              "type_marker": null
            },
            "schmitt_trigger": false,
            "termination_enabled": false,
            "uses_data_gate": false
          }
        },
        "name": "$auto$iopadmap.cc:369:execute$296",
        "location": {
          "fb": 1,
          "i": 4
        }
      },
      {
        "variant": {
          "IOBuf": {
            "input": {
              "i": 22,
              "type_marker": null
            },
            "oe": null,
            "output": null,
            "schmitt_trigger": false,
            "termination_enabled": false,
            "slew_is_fast": false,
            "uses_data_gate": false
          }
        },
        "name": "$auto$iopadmap.cc:369:execute$297",
        "location": {
          "fb": 0,
          "i": 1
        }
      },
      {
        "variant": {
          "IOBuf": {
            "input": {
              "i": 23,
              "type_marker": null
            },
            "oe": null,
            "output": null,
            "schmitt_trigger": false,
            "termination_enabled": false,
            "slew_is_fast": false,
            "uses_data_gate": false
          }
        },
        "name": "$auto$iopadmap.cc:369:execute$298",
        "location": {
          "fb": 0,
          "i": 5
        }
      },
      {
        "variant": {
          "Reg": {
            "mode": "DFF",
            "clkinv": false,
            "clkddr": false,
            "init_state": false,
            "set_input": {
              "i": 1,
              "type_marker": null
            },
            "reset_input": {
              "i": 1,
              "type_marker": null
            },
            "ce_input": null,
            "dt_input": {
              "i": 18,
              "type_marker": null
            },
            "clk_input": {
              "i": 24,
              "type_marker": null
            },
            "output": {
              "i": 2,
              "type_marker": null
            }
          }
        },
        "name": "$auto$simplemap.cc:420:simplemap_dff$154",
        "location": null
      },
      {
        "variant": {
          "Reg": {
            "mode": "LATCH",
            "clkinv": true,
            "clkddr": false,
            "init_state": false,
            "set_input": {
              "i": 1,
              "type_marker": null
            },
            "reset_input": {
              "i": 1,
              "type_marker": null
            },
            "ce_input": null,
            "dt_input": {
              "i": 16,
              "type_marker": null
            },
            "clk_input": {
              "i": 14,
              "type_marker": null
            },
            "output": {
              "i": 22,
              "type_marker": null
            }
          }
        },
        "name": "$auto$simplemap.cc:517:simplemap_dlatch$152",
        "location": null
      },
      {
        "variant": {
          "Reg": {
            "mode": "LATCH",
            "clkinv": true,
            "clkddr": false,
            "init_state": false,
            "set_input": {
              "i": 1,
              "type_marker": null
            },
            "reset_input": {
              "i": 1,
              "type_marker": null
            },
            "ce_input": null,
            "dt_input": {
              "i": 20,
              "type_marker": null
            },
            "clk_input": {
              "i": 14,
              "type_marker": null
            },
            "output": {
              "i": 23,
              "type_marker": null
            }
          }
        },
        "name": "$auto$simplemap.cc:517:simplemap_dlatch$153",
        "location": null
      },
      {
        "variant": {
          "Reg": {
            "mode": "TFF",
            "clkinv": false,
            "clkddr": false,
            "init_state": false,
            "set_input": {
              "i": 1,
              "type_marker": null
            },
            "reset_input": {
              "i": 1,
              "type_marker": null
            },
            "ce_input": null,
            "dt_input": {
              "i": 4,
              "type_marker": null
            },
            "clk_input": {
              "i": 24,
              "type_marker": null
            },
            "output": {
              "i": 6,
              "type_marker": null
            }
          }
        },
        "name": "$extract$\\FTCP$262",
        "location": null
      },
      {
        "variant": {
          "Reg": {
            "mode": "TFF",
            "clkinv": false,
            "clkddr": false,
            "init_state": false,
            "set_input": {
              "i": 1,
              "type_marker": null
            },
            "reset_input": {
              "i": 1,
              "type_marker": null
            },
            "ce_input": null,
            "dt_input": {
              "i": 17,
              "type_marker": null
            },
            "clk_input": {
              "i": 24,
              "type_marker": null
            },
            "output": {
              "i": 5,
              "type_marker": null
            }
          }
        },
        "name": "$extract$\\FTCP$263",
        "location": null
      },
      {
        "variant": {
          "BufgClk": {
            "input": {
              "i": 21,
              "type_marker": null
            },
            "output": {
              "i": 24,
              "type_marker": null
            }
          }
        },
        "name": "bufg0",
        "location": null
      }
    ]
  },
  "nets": {
    "storage": [
      {
        "name": "<internal virtual Vdd net>",
        "source": null,
        "sinks": []
      },
      {
        "name": "<internal virtual Vss net>",
        "source": null,
        "sinks": [
          {
            "i": 19,
            "type_marker": null
          },
          {
            "i": 19,
            "type_marker": null
          },
          {
            "i": 20,
            "type_marker": null
          },
          {
            "i": 20,
            "type_marker": null
          },
          {
            "i": 21,
            "type_marker": null
          },
          {
            "i": 21,
            "type_marker": null
          },
          {
            "i": 22,
            "type_marker": null
          },
          {
            "i": 22,
            "type_marker": null
          },
          {
            "i": 23,
            "type_marker": null
          },
          {
            "i": 23,
            "type_marker": null
          }
        ]
      },
      {
        "name": "dotcnt[0]",
        "source": {
          "i": 19,
          "type_marker": null
        },
        "sinks": [
          {
            "i": 0,
            "type_marker": null
          },
          {
            "i": 6,
            "type_marker": null
          },
          {
            "i": 8,
            "type_marker": null
          }
        ]
      },
      {
        "name": "$auto$coolrunner2_fixup.cc:98:execute$303",
        "source": {
          "i": 0,
          "type_marker": null
        },
        "sinks": [
          {
            "i": 1,
            "type_marker": null
          }
        ]
      },
      {
        "name": "$auto$coolrunner2_fixup.cc:99:execute$304",
        "source": {
          "i": 1,
          "type_marker": null
        },
        "sinks": [
          {
            "i": 22,
            "type_marker": null
          }
        ]
      },
      {
        "name": "dotcnt[2]",
        "source": {
          "i": 23,
          "type_marker": null
        },
        "sinks": [
          {
            "i": 2,
            "type_marker": null
          },
          {
            "i": 5,
            "type_marker": null
          },
          {
            "i": 7,
            "type_marker": null
          }
        ]
      },
      {
        "name": "dotcnt[1]",
        "source": {
          "i": 22,
          "type_marker": null
        },
        "sinks": [
          {
            "i": 2,
            "type_marker": null
          },
          {
            "i": 4,
            "type_marker": null
          },
          {
            "i": 6,
            "type_marker": null
          },
          {
            "i": 7,
            "type_marker": null
          }
        ]
      },
      {
        "name": "$auto$coolrunner2_sop.cc:124:execute$273",
        "source": {
          "i": 2,
          "type_marker": null
        },
        "sinks": [
          {
            "i": 9,
            "type_marker": null
          }
        ]
      },
      {
        "name": "$abc$264$new_n11_",
        "source": {
          "i": 9,
          "type_marker": null
        },
        "sinks": [
          {
            "i": 3,
            "type_marker": null
          }
        ]
      },
      {
        "name": "$auto$coolrunner2_sop.cc:124:execute$276",
        "source": {
          "i": 3,
          "type_marker": null
        },
        "sinks": [
          {
            "i": 10,
            "type_marker": null
          }
        ]
      },
      {
        "name": "$auto$iopadmap.cc:409:execute$302",
        "source": {
          "i": 15,
          "type_marker": null
        },
        "sinks": [
          {
            "i": 4,
            "type_marker": null
          },
          {
            "i": 5,
            "type_marker": null
          }
        ]
      },
      {
        "name": "$auto$coolrunner2_sop.cc:124:execute$279",
        "source": {
          "i": 4,
          "type_marker": null
        },
        "sinks": [
          {
            "i": 13,
            "type_marker": null
          }
        ]
      },
      {
        "name": "$auto$coolrunner2_sop.cc:124:execute$281",
        "source": {
          "i": 5,
          "type_marker": null
        },
        "sinks": [
          {
            "i": 13,
            "type_marker": null
          }
        ]
      },
      {
        "name": "$auto$coolrunner2_sop.cc:124:execute$286",
        "source": {
          "i": 6,
          "type_marker": null
        },
        "sinks": [
          {
            "i": 11,
            "type_marker": null
          }
        ]
      },
      {
        "name": "$auto$coolrunner2_sop.cc:124:execute$289",
        "source": {
          "i": 7,
          "type_marker": null
        },
        "sinks": [
          {
            "i": 20,
            "type_marker": null
          },
          {
            "i": 21,
            "type_marker": null
          }
        ]
      },
      {
        "name": "$auto$coolrunner2_sop.cc:124:execute$292",
        "source": {
          "i": 8,
          "type_marker": null
        },
        "sinks": [
          {
            "i": 12,
            "type_marker": null
          }
        ]
      },
      {
        "name": "$0\\r0[0:0]",
        "source": {
          "i": 10,
          "type_marker": null
        },
        "sinks": [
          {
            "i": 20,
            "type_marker": null
          }
        ]
      },
      {
        "name": "$abc$249$new_n18_",
        "source": {
          "i": 11,
          "type_marker": null
        },
        "sinks": [
          {
            "i": 23,
            "type_marker": null
          }
        ]
      },
      {
        "name": "$0\\dotcnt[2:0][0]",
        "source": {
          "i": 12,
          "type_marker": null
        },
        "sinks": [
          {
            "i": 19,
            "type_marker": null
          }
        ]
      },
      {
        "name": "$auto$coolrunner2_sop.cc:214:execute$283",
        "source": {
          "i": 13,
          "type_marker": null
        },
        "sinks": [
          {
            "i": 14,
            "type_marker": null
          }
        ]
      },
      {
        "name": "$0\\r1[0:0]",
        "source": {
          "i": 14,
          "type_marker": null
        },
        "sinks": [
          {
            "i": 21,
            "type_marker": null
          }
        ]
      },
      {
        "name": "$auto$iopadmap.cc:409:execute$301",
        "source": {
          "i": 16,
          "type_marker": null
        },
        "sinks": [
          {
            "i": 24,
            "type_marker": null
          }
        ]
      },
      {
        "name": "$auto$iopadmap.cc:409:execute$300",
        "source": {
          "i": 20,
          "type_marker": null
        },
        "sinks": [
          {
            "i": 17,
            "type_marker": null
          }
        ]
      },
      {
        "name": "$auto$iopadmap.cc:409:execute$299",
        "source": {
          "i": 21,
          "type_marker": null
        },
        "sinks": [
          {
            "i": 18,
            "type_marker": null
          }
        ]
      },
      {
        "name": "clk",
        "source": {
          "i": 24,
          "type_marker": null
        },
        "sinks": [
          {
            "i": 19,
            "type_marker": null
          },
          {
            "i": 22,
            "type_marker": null
          },
          {
            "i": 23,
            "type_marker": null
          }
        ]
      }
    ]
  },
  "vdd_net": {
    "i": 0,
    "type_marker": null
  },
  "vss_net": {
    "i": 1,
    "type_marker": null
  }
}
