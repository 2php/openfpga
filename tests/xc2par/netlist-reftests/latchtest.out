{
  "mcs": {
    "storage": [
      {
        "name": "$auto$coolrunner2_sop.cc:154:execute$278_$auto$simplemap.cc:517:simplemap_dlatch$152_$auto$iopadmap.cc:369:execute$297",
        "requested_loc": {
          "fb": 0,
          "i": 1
        },
        "io_bits": {
          "input": "Reg",
          "oe": null,
          "schmitt_trigger": false,
          "termination_enabled": false,
          "slew_is_fast": false,
          "uses_data_gate": false
        },
        "reg_bits": {
          "mode": "LATCH",
          "clkinv": true,
          "clkddr": false,
          "init_state": false,
          "set_input": null,
          "reset_input": null,
          "ce_input": null,
          "dt_input": "Xor",
          "clk_input": {
            "PTerm": {
              "i": 5,
              "type_marker": null
            }
          }
        },
        "xor_bits": {
          "orterm_inputs": [],
          "andterm_input": {
            "i": 4,
            "type_marker": null
          },
          "invert_out": false
        },
        "io_feedback_used": false,
        "reg_feedback_used": false,
        "xor_feedback_used": false
      },
      {
        "name": "$auto$coolrunner2_sop.cc:223:execute$285_$auto$simplemap.cc:517:simplemap_dlatch$153_$auto$iopadmap.cc:369:execute$298",
        "requested_loc": {
          "fb": 0,
          "i": 5
        },
        "io_bits": {
          "input": "Reg",
          "oe": null,
          "schmitt_trigger": false,
          "termination_enabled": false,
          "slew_is_fast": false,
          "uses_data_gate": false
        },
        "reg_bits": {
          "mode": "LATCH",
          "clkinv": true,
          "clkddr": false,
          "init_state": false,
          "set_input": null,
          "reset_input": null,
          "ce_input": null,
          "dt_input": "Xor",
          "clk_input": {
            "PTerm": {
              "i": 8,
              "type_marker": null
            }
          }
        },
        "xor_bits": {
          "orterm_inputs": [
            {
              "i": 6,
              "type_marker": null
            },
            {
              "i": 7,
              "type_marker": null
            }
          ],
          "andterm_input": null,
          "invert_out": false
        },
        "io_feedback_used": false,
        "reg_feedback_used": false,
        "xor_feedback_used": false
      },
      {
        "name": "$auto$coolrunner2_fixup.cc:108:execute$306_$extract$\\FTCP$262",
        "requested_loc": null,
        "io_bits": null,
        "reg_bits": {
          "mode": "TFF",
          "clkinv": false,
          "clkddr": false,
          "init_state": false,
          "set_input": null,
          "reset_input": null,
          "ce_input": null,
          "dt_input": "Xor",
          "clk_input": {
            "GCK": {
              "i": 0,
              "type_marker": null
            }
          }
        },
        "xor_bits": {
          "orterm_inputs": [],
          "andterm_input": {
            "i": 1,
            "type_marker": null
          },
          "invert_out": false
        },
        "io_feedback_used": false,
        "reg_feedback_used": true,
        "xor_feedback_used": false
      },
      {
        "name": "$auto$coolrunner2_sop.cc:154:execute$275",
        "requested_loc": null,
        "io_bits": null,
        "reg_bits": null,
        "xor_bits": {
          "orterm_inputs": [],
          "andterm_input": {
            "i": 3,
            "type_marker": null
          },
          "invert_out": false
        },
        "io_feedback_used": false,
        "reg_feedback_used": false,
        "xor_feedback_used": true
      },
      {
        "name": "$auto$coolrunner2_sop.cc:154:execute$288_$extract$\\FTCP$263",
        "requested_loc": null,
        "io_bits": null,
        "reg_bits": {
          "mode": "TFF",
          "clkinv": false,
          "clkddr": false,
          "init_state": false,
          "set_input": null,
          "reset_input": null,
          "ce_input": null,
          "dt_input": "Xor",
          "clk_input": {
            "GCK": {
              "i": 0,
              "type_marker": null
            }
          }
        },
        "xor_bits": {
          "orterm_inputs": [],
          "andterm_input": {
            "i": 2,
            "type_marker": null
          },
          "invert_out": false
        },
        "io_feedback_used": false,
        "reg_feedback_used": true,
        "xor_feedback_used": false
      },
      {
        "name": "$auto$coolrunner2_sop.cc:154:execute$294_$auto$simplemap.cc:420:simplemap_dff$154",
        "requested_loc": null,
        "io_bits": null,
        "reg_bits": {
          "mode": "DFF",
          "clkinv": false,
          "clkddr": false,
          "init_state": false,
          "set_input": null,
          "reset_input": null,
          "ce_input": null,
          "dt_input": "Xor",
          "clk_input": {
            "GCK": {
              "i": 0,
              "type_marker": null
            }
          }
        },
        "xor_bits": {
          "orterm_inputs": [],
          "andterm_input": {
            "i": 0,
            "type_marker": null
          },
          "invert_out": false
        },
        "io_feedback_used": false,
        "reg_feedback_used": true,
        "xor_feedback_used": false
      },
      {
        "name": "$auto$iopadmap.cc:369:execute$295",
        "requested_loc": {
          "fb": 0,
          "i": 13
        },
        "io_bits": {
          "input": null,
          "oe": null,
          "schmitt_trigger": false,
          "termination_enabled": false,
          "slew_is_fast": true,
          "uses_data_gate": false
        },
        "reg_bits": null,
        "xor_bits": null,
        "io_feedback_used": true,
        "reg_feedback_used": false,
        "xor_feedback_used": false
      },
      {
        "name": "$auto$iopadmap.cc:369:execute$296",
        "requested_loc": {
          "fb": 1,
          "i": 4
        },
        "io_bits": {
          "input": null,
          "oe": null,
          "schmitt_trigger": false,
          "termination_enabled": false,
          "slew_is_fast": true,
          "uses_data_gate": false
        },
        "reg_bits": null,
        "xor_bits": null,
        "io_feedback_used": false,
        "reg_feedback_used": false,
        "xor_feedback_used": false
      }
    ]
  },
  "pterms": {
    "storage": [
      {
        "name": "$auto$coolrunner2_sop.cc:143:execute$293",
        "requested_loc": null,
        "inputs_true": [],
        "inputs_comp": [
          [
            "Reg",
            {
              "i": 5,
              "type_marker": null
            }
          ]
        ]
      },
      {
        "name": "$auto$coolrunner2_fixup.cc:101:execute$305",
        "requested_loc": null,
        "inputs_true": [
          [
            "Reg",
            {
              "i": 5,
              "type_marker": null
            }
          ]
        ],
        "inputs_comp": []
      },
      {
        "name": "$auto$coolrunner2_sop.cc:143:execute$287",
        "requested_loc": null,
        "inputs_true": [
          [
            "Reg",
            {
              "i": 2,
              "type_marker": null
            }
          ],
          [
            "Reg",
            {
              "i": 5,
              "type_marker": null
            }
          ]
        ],
        "inputs_comp": []
      },
      {
        "name": "$auto$coolrunner2_sop.cc:143:execute$274",
        "requested_loc": null,
        "inputs_true": [
          [
            "Reg",
            {
              "i": 4,
              "type_marker": null
            }
          ],
          [
            "Reg",
            {
              "i": 2,
              "type_marker": null
            }
          ]
        ],
        "inputs_comp": []
      },
      {
        "name": "$auto$coolrunner2_sop.cc:143:execute$277",
        "requested_loc": null,
        "inputs_true": [],
        "inputs_comp": [
          [
            "Xor",
            {
              "i": 3,
              "type_marker": null
            }
          ]
        ]
      },
      {
        "name": "$auto$coolrunner2_sop.cc:143:execute$290",
        "requested_loc": null,
        "inputs_true": [],
        "inputs_comp": [
          [
            "Reg",
            {
              "i": 4,
              "type_marker": null
            }
          ],
          [
            "Reg",
            {
              "i": 2,
              "type_marker": null
            }
          ]
        ]
      },
      {
        "name": "$auto$coolrunner2_sop.cc:143:execute$282",
        "requested_loc": null,
        "inputs_true": [],
        "inputs_comp": [
          [
            "Reg",
            {
              "i": 4,
              "type_marker": null
            }
          ],
          [
            "Pin",
            {
              "i": 6,
              "type_marker": null
            }
          ]
        ]
      },
      {
        "name": "$auto$coolrunner2_sop.cc:143:execute$280",
        "requested_loc": null,
        "inputs_true": [],
        "inputs_comp": [
          [
            "Reg",
            {
              "i": 2,
              "type_marker": null
            }
          ],
          [
            "Pin",
            {
              "i": 6,
              "type_marker": null
            }
          ]
        ]
      },
      {
        "name": "$auto$coolrunner2_sop.cc:143:execute$290",
        "requested_loc": null,
        "inputs_true": [],
        "inputs_comp": [
          [
            "Reg",
            {
              "i": 4,
              "type_marker": null
            }
          ],
          [
            "Reg",
            {
              "i": 2,
              "type_marker": null
            }
          ]
        ]
      }
    ]
  },
  "bufg_clks": {
    "storage": [
      {
        "name": "bufg0",
        "requested_loc": null,
        "input": {
          "i": 7,
          "type_marker": null
        }
      }
    ]
  },
  "bufg_gts": {
    "storage": []
  },
  "bufg_gsr": {
    "storage": []
  }
}
